### PR TITLE
kanata: add support for darwin platforms

### PR DESCRIPTION
**Pull request by bmanuel**
Darwin support was added upstream in v1.5.0. See https://github.com/jtroo/kanata/releases/tag/v1.5.0.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (jian-lin) [inline pkgs/top-level/all-packages.nix:1934]: This change is unnecessary and should be removed.

File: pkgs/tools/system/kanata/default.nix
Comment (viraptor) [inline pkgs/tools/system/kanata/default.nix:40]: if you want linux+darwin, that's `platforms.unix`.

File: pkgs/tools/system/kanata/default.nix
Comment (viraptor) [inline pkgs/tools/system/kanata/default.nix:5]: Consider including darwin instead. While inheriting specific packages was the thing to do historically, this will prevent moving the package to the `by-name` space in the future.

File: pkgs/tools/system/kanata/default.nix
Comment (jian-lin) [reply pkgs/tools/system/kanata/default.nix:40]: Upstream supports windows as well.  However, I am not sure how to deal with windows in Nixpkgs.

On `unix` vs `all`, I find that the manual [says][1]:

> In general it is preferable to set `meta.platforms = lib.platforms.all` and then exclude (using `badPlatforms`) any platforms on which the package is known not to build

[1]: https://nixos.org/manual/nixpkgs/unstable/#var-meta-badPlatforms

File: pkgs/tools/system/kanata/default.nix
Comment (jian-lin) [reply pkgs/tools/system/kanata/default.nix:5]: > Consider including darwin instead.

What do you mean by that?  Is [this][1] the comment you refer to?

[1]: https://github.com/NixOS/nixpkgs/pull/281437/files#r1458023659

File: pkgs/tools/system/kanata/default.nix
Comment (viraptor) [reply pkgs/tools/system/kanata/default.nix:40]: Ah, from the PR description I got just "added darwin". If others are supported, then all good 👍

File: pkgs/tools/system/kanata/default.nix
Comment (viraptor) [reply pkgs/tools/system/kanata/default.nix:5]: Instead of inheriting `IOKit` itself in all-packages, add darwin to arguments and either add the whole `darwin.apple_sdk.frameworks.IOKit` to dependencies, or `let` assign the name like in https://github.com/NixOS/nixpkgs/blob/4e1582c0136f371f68c23074acf8ae22ddb14a0a/pkgs/by-name/cm/cmake/package.nix#L35

This allows the main package to be used without the extra IOKit option, so we'll be able to move it to the `by-name` area.

File: pkgs/tools/system/kanata/default.nix
Comment (bmanuel) [reply pkgs/tools/system/kanata/default.nix]: I intended to only add darwin support since I don't have any way to test that windows can actually build the package. I am happy to leave `platforms.all`, but I don't want to increase the support burden if it will be completely untested.

**Review by viraptor — APPROVED** [review]
Just waiting for the CI builds to finish.

File: pkgs/tools/system/kanata/default.nix
Comment (jian-lin) [inline pkgs/tools/system/kanata/default.nix:39]: Would you like to add yourself to the maintainer list?  This makes it easier for you to maintain this package.  For example, our CI will [ping][1] you when it creates a PR to bump the version.


Here is the [doc][2] and an [example][3] of doing so.

[1]: https://github.com/NixOS/nixpkgs/pull/276116
[2]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[3]: https://github.com/NixOS/nixpkgs/commit/7f10f172110477ea263d63b6c793ebd8637eed63

Comment (bmanuel) [reply pkgs/tools/system/kanata/default.nix:39]: I am definitely up for helping. Is it best to do that in a separate PR or as part of this one?

Comment (jian-lin) [reply pkgs/tools/system/kanata/default.nix:39]: You can do that in this PR with two commits:

- in the first commit, you add your entry to `maintainer-list.nix`
- in the second commit, you change the current commit by adding yourself to `maintainers` list.

**Review by iFreilicht — APPROVED** [review]

**Pull request merged**
